j2cl-maven-plugin One or more test(s) failed message improvements

```
j2cl-maven-plugin One or more test(s) failed

       j2cl-maven-plugin test runner [FAILED]
[ERROR] walkingkooka:walkingkooka-net-http-client-it-junit-test:jar:1.0-SNAPSHOT-JUNIT_WEBDRIVER_TESTS                 /Users/miroslav/repos-github/walkingkooka-net-http-client/src/it/junit-test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--walkingkooka-net-http-client-it-junit-test--jar--1.0-SNAPSHOT-40e50f9be125f925a2557688a33fbf5509714abb/8-closure-compile/output/test.JunitTest.html
[ERROR] walkingkooka:walkingkooka-net-http-client-it-junit-test:jar:1.0-SNAPSHOT-JUNIT_WEBDRIVER_TESTS                 9 of 9 tests run in 868.1100000394508ms.
[ERROR] walkingkooka:walkingkooka-net-http-client-it-junit-test:jar:1.0-SNAPSHOT-JUNIT_WEBDRIVER_TESTS                 7 passed, 2 failed.
[ERROR] walkingkooka:walkingkooka-net-http-client-it-junit-test:jar:1.0-SNAPSHOT-JUNIT_WEBDRIVER_TESTS                 96 ms/test. 1 files loaded.
```

Message should say how many tests failed out of the total.

```
 Test Report
j2cl-maven-plugin test runner [FAILED]
/Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.html
4 of 4 tests run in 5.874999973457307ms.
2 passed, 2 failed.
1 ms/test. 1 files loaded.
ERROR in testSimple3
org.junit.ComparisonFailure: expected:<Transitive123[*******]> but was:<Transitive123[]>
Error
    at zd (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:79:186)
    at X.qa (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:79:682)
    at P.m.M (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:63:56)
    at P.m.la (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:61:446)
    at V (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:77:36)
    at P.<anonymous> (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:58:414)
    at V (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:77:36)
    at P.<anonymous> (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:58:302)
    at new F (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:13:649)
    at fd (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:58:284)
    at ae (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:88:423)
    at file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:96:14
ERROR in testSimple4
org.junit.ComparisonFailure: expected:<Transitive123[21111]> but was:<Transitive123[]>
Error
    at zd (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:79:186)
    at X.ra (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:79:726)
    at P.m.M (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:63:56)
    at P.m.la (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:61:446)
    at V (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:77:36)
    at P.<anonymous> (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:58:414)
    at V (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:77:36)
    at P.<anonymous> (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:58:302)
    at new F (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:13:649)
    at fd (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:58:284)
    at ae (file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:88:423)
    at file:///Users/miroslav/repos-github/j2cl-maven-plugin/target/it-tests/junit-test-dependency-graph/test/target/walkingkooka-j2cl-maven-plugin-cache/walkingkooka--j2cl-maven-plugin-it-junit-test-dependency-graph-test--jar--1.0-SNAPSHOT-23ba0b683c80a96667591843b0e6c77a20ff04f2/8-closure-compile/output/test.JunitTest.js:96:14               
```

--- a/src/main/java/walkingkooka/j2cl/maven/J2clStepWorkerWebDriverUnitTestRunner.java
+++ b/src/main/java/walkingkooka/j2cl/maven/J2clStepWorkerWebDriverUnitTestRunner.java
@@ -25,8 +25,10 @@ import org.openqa.selenium.logging.LogType;
 import org.openqa.selenium.support.ui.FluentWait;
 import walkingkooka.text.CharSequences;
 
+import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.StringReader;
 import java.nio.charset.Charset;
 import java.time.Duration;
 import java.util.List;
@@ -125,16 +127,17 @@ final class J2clStepWorkerWebDriverUnitTestRunner extends J2clStepWorker2 {
                                 .pollingEvery(Duration.ofMillis(100))
                                 .until(d -> isFinished(d));
 
+                        final String testReport = testReport(driver);
                         logger.indent();
                         {
-                            printTestReport(logger, driver);
+                            printTestReport(logger, testReport);
                             printBrowserLogs(driver, logLevel, logger);
                         }
                         logger.outdent();
 
                         // check for success
                         if (!isSuccess(driver)) {
-                            throw new J2clException("One or more test(s) failed");
+                            throw new J2clException(testsFailedMessage(testReport));
                         }
                         logger.printLine("All test(s) successful!");
                     } catch (final J2clException rethrow) {
@@ -155,8 +158,40 @@ final class J2clStepWorkerWebDriverUnitTestRunner extends J2clStepWorker2 {
         logger.outdent();
     }
 
+    private static String testReport(final WebDriver driver) {
+        return executeScript(driver, "return window.G_testRunner.getReport()");
+    }
+
     private static boolean isSuccess(final WebDriver driver) {
         return executeScript(driver, "return window.G_testRunner.isSuccess()");
+    }
+
+    /**
+     * <pre>
+     * 4 of 4 tests run in 5.874999973457307ms.
+     * 2 passed, 2 failed.
+     * 1 ms/test. 1 files loaded.
+     * ERROR in testSimple3
+     * </pre>
+     */
+    private static String testsFailedMessage(final String testReport) throws IOException {
+        String message = "One or more test(s) failed";
+
+        try (final BufferedReader reader = new BufferedReader(new StringReader(testReport))) {
+            for (; ; ) {
+                final String line = reader.readLine();
+                if (null == line) {
+                    break; // EOF
+                }
+
+                if (line.contains("failed")) {
+                    message = line.trim();
+                    break;
+                }
+            }
+        }
+
+        return message;
     }
 
     private static boolean isFinished(final WebDriver driver) {
@@ -168,11 +203,12 @@ final class J2clStepWorkerWebDriverUnitTestRunner extends J2clStepWorker2 {
         return (T) ((JavascriptExecutor) driver).executeScript(script);
     }
 
-    private void printTestReport(J2clLinePrinter logger, WebDriver driver) {
+    private static void printTestReport(final J2clLinePrinter logger,
+                                        final String testReport) {
         try {
             logger.printLine("Test Report");
             logger.indent();
-            logger.printLine(executeScript(driver, "return window.G_testRunner.getReport()"));
+            logger.printLine(testReport);
         } finally {
             logger.outdent();
         }

